File: app-ios/ios/service/ArticleService.m
//
//  ArticleService.m
//  ios
//
//  Created by Peter Zhan on 2017/10/16.
//  Copyright © 2017年 com.enuocms. All rights reserved.
//

#import "ArticleService.h"

@implementation ArticleService
-(void)pageList:(NSInteger)pageIndex
       pageSize:(NSInteger)paegSize
        success:(APIPageDataHandle)success_callback
        failure:(APIErrorHandle)failure_callback{
    
    NSMutableDictionary *parameters = [self UserInfoparameters];
    [parameters setObject:[NSString stringWithFormat:@"%ld" , pageIndex]  forKey:@"pageIndex"];
    [parameters setObject:[NSString stringWithFormat:@"%ld" , paegSize]  forKey:@"paegSize"];
    
    [self GetRequestWithFunction:@"article/list" parameters:parameters success:^(APIResult *result, NSDictionary *dict) {
        
        APIPageResult *page = [[APIPageResult alloc] init];
        if(result.success){
            NSDictionary *data = [dict objectForKey:@"data"];
            page.total = [data intForKey:@"total"];
            NSMutableArray *arry = [data objectForKey:@"items"];
            for (NSDictionary *item  in arry) {
                Article *ent = [item toModel:[Article class]];
                ent.ID = [item stringForKey:@"id"];
                [page.items addObject:ent];
            }
        }
        
        if (success_callback) {
            success_callback(result,page);
        }
    } failure:failure_callback];
}
@end

File: app-swift/ios/ios/service/ArticleService.swift
//
//  ArticleService.swift
//  ios
//
//  Created by Peter Zhan on 2017/10/20.
//  Copyright © 2017年 com.enuocms. All rights reserved.
//

import Foundation

class ArticleService: BaseService {
    func pagelist(pageIndex:Int,pageSize:Int,success:@escaping APIPageDataHandle,failure:@escaping APIErrorHandle) {
        var parameters = self.UserInfoparameters()
        parameters["pageIndex"] = String(pageIndex)
        parameters["pageSize"] = String(pageSize)
        
        self.RequestWithFunction(url: "article/list", parameters: parameters, method: .get, successHandle: { (result, json) in
            let pageList = ApiPageResult.init()
            pageList.items = []
            pageList.total = pageSize
            pageList.pageIndex = pageIndex
            pageList.pageSize = pageSize
            if(result.success){
                let data = json["data"]
                pageList.total = data["total"].intValue
                let array = data["items"].arrayValue
                for item in array{
                    let ent = Article.deserialize(from: item.rawString())
                    pageList.items.append(ent!)
                }
            }
            success(result,pageList)
        }, failureHandle: failure)
    }
}
